File: Assignment_032/Program4.java
import java.util.Scanner;

class Bitwise
{
    public int ToggleBit(int iNo, int iPos)
    {
        int iMask = 1;
        int iResult = 0;

        iMask = iMask << iPos-1;
        iResult = iNo ^ iMask;

       return iResult;
    }
}
class Program4
{
    public static void main(String A[])
    {
        int iValue = 0;
        int iLocation = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the Number : ");
        iValue = sobj.nextInt();

        System.out.println("Enter the Position : ");
        iLocation = sobj.nextInt();

        Bitwise bobj = new Bitwise();

        int iRet = 0;

        iRet = bobj.ToggleBit(iValue,iLocation);

        System.out.println("Updated Number is : "+iRet);

    }
}

File: Assignment_042/program4.c
#include<stdio.h>


int Factorial(int iNo)
{
    static int iFact = 1;

    if(iNo > 1)
    {
        iFact = iFact * iNo;

        Factorial(--iNo);
    }

    return iFact;
}
int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter a number : \n");
    scanf("%d",&iValue);

    iRet = Factorial(iValue);

    printf("Factorial is : %d\n",iRet);


    return 0;
}